persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/datastore/InternalDataStore.java

```diff
@@ -303,28 +302,22 @@ else if (null != batchStatements)
             {
                 try
                 {
-                    long queryStartNanoTime = System.nanoTime();
-                    CQLStatement statement = QueryProcessor.parseStatement(unpreparedCql, queryState).statement;
-                    if (!StargateSystemKeyspace.maybeCompleteSystemLocalOrPeersInternal(statement, queryState, queryOptions, queryStartNanoTime, future))
-                    {
-                        ResultMessage resultMessage = QueryProcessor.instance.processStatement(statement, queryState, queryOptions, queryStartNanoTime);
-                        if (resultMessage instanceof ResultMessage.Rows) {
-                            try {
-                                org.apache.cassandra.cql3.ResultSet result = ((ResultMessage.Rows) resultMessage).result;
-                                // Didn't really want to grab this using reflection but the only alternative would be to decode
-                                // the ResultMessage in the same way the driver does which is overkill in order to pull a single
-                                // field. Luckily this is public in both DSE and C* 4.0 so we'll only need this here.
-                                Field f = result.metadata.getClass().getDeclaredField("pagingState");
-                                f.setAccessible(true);
-                                this.pagingState = (PagingState) f.get(result.metadata);
-                            } catch (Exception e)
-                            {
-                                LOG.info("Unable to get paging state", e);
-                            }
+                    ResultMessage resultMessage = QueryProcessor.instance.process(unpreparedCql, queryState, queryOptions, System.nanoTime());
+                    if (resultMessage instanceof ResultMessage.Rows) {
+                        try {
+                            org.apache.cassandra.cql3.ResultSet result = ((ResultMessage.Rows) resultMessage).result;
+                            // Didn't really want to grab this using reflection but the only alternative would be to decode
+                            // the ResultMessage in the same way the driver does which is overkill in order to pull a single
+                            // field. Luckily this is public in both DSE and C* 4.0 so we'll only need this here.
+                            Field f = result.metadata.getClass().getDeclaredField("pagingState");
+                            f.setAccessible(true);
+                            this.pagingState = (PagingState) f.get(result.metadata);
+                        } catch (Exception e)
+                        {
+                            LOG.info("Unable to get paging state", e);
```

Comment:
Can we extract this common logic into a method and reuse it below?

Reply:
Good idea.